pkg/util/stmtsummary/statement_summary_test.go

```diff
@@ -1563,4 +1563,32 @@ func TestAccessPrivilege(t *testing.T) {
 	reader.hasProcessPriv = true
 	datums = reader.GetStmtSummaryHistoryRows()
 	require.Len(t, datums, loops)
+
+	// Test the same query digests, but run as a different user in a new statement
+	// summary interval. The old user should not be able to access the rows generated
+	// for the new user.
+	ssMap.beginTimeForCurInterval = time.Now().Unix()
```

Comment:
Is this required?

Reply:
I think it is. The intent here is to create a new statement summary interval with a new, empty `authUsers` set, because `authUsers` represents the union of all users who have executed this query for a given interval. If we use the same interval, then `authUsers` will include both users.

More broadly speaking, testing with a new interval makes sure that a user cannot read rows from the current statement summary that they are not authorized to access, even if they ran the same query in the past. (This is what the bug would have allowed, since it was effectively not resetting `authUsers` for new statement summaries.)